client/pages/user/[username].tsx

```diff
@@ -0,0 +1,123 @@
+/** @jsx jsx */
+
+import React from 'react';
+import { jsx } from '@emotion/core';
+import { NextPage } from 'next';
+import Head from 'next/head';
+import { useTranslation } from 'i18n';
+import {
+  getTitle,
+  getUserProfileMetaImage,
+  getUserProfileMetaBuildCount,
+} from 'common/utils';
+import UserProfile from 'components/common/UserProfile';
+import { useRouter } from 'next/router';
+import { Media } from 'components/common/Media';
+import DesktopLayout from 'components/desktop/Layout';
+import MobileLayout from 'components/mobile/Layout';
+import { useQuery } from '@apollo/client';
+import userProfileQuery from 'graphql/queries/userProfile.graphql';
+import currentUserQuery from 'graphql/queries/currentUser.graphql';
+import {
+  userProfile,
+  userProfileVariables,
+} from 'graphql/queries/__generated__/userProfile';
+import { currentUser } from 'graphql/queries/__generated__/currentUser';
+import ErrorPage from 'pages/_error';
+
+const UserProfilePage: NextPage = () => {
+  const { t } = useTranslation('common');
+
+  const router = useRouter();
+  const username = Array.isArray(router.query.username)
+    ? router.query.username[0]
+    : router.query.username!;
```

Comment:
Not a huge deal, but I'd avoid this by just throwing an error if it doesn't exist.